app/ui/legacy/src/main/java/com/fsck/k9/activity/MessageRepository.java

```diff
@@ -0,0 +1,84 @@
+package com.fsck.k9.activity;
+
+import android.content.Context;
+import android.os.Bundle;
+
+import androidx.fragment.app.Fragment;
+import androidx.loader.app.LoaderManager;
+import androidx.loader.app.LoaderManager.LoaderCallbacks;
+import androidx.loader.content.Loader;
+import com.fsck.k9.Account;
+import com.fsck.k9.Preferences;
+import com.fsck.k9.controller.MessageReference;
+import com.fsck.k9.controller.MessagingController;
+import com.fsck.k9.mailstore.LocalMessage;
+import com.fsck.k9.ui.message.LocalMessageLoader;
+import timber.log.Timber;
+
+public abstract class MessageRepository {
```

Comment:
Forget about trying to reuse any of the existing code. For rough guidance take a look at `FolderRepository`. For the loading/UI logic take a look at `ChangelogViewModel`.

If you don't want to deal with writing `MessageRepository` to access the database, let me know and I'll do it.